showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITCrud.java

```diff
@@ -0,0 +1,159 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.showcase.v1beta1.it;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.protobuf.FieldMask;
+import com.google.showcase.v1beta1.CreateUserRequest;
+import com.google.showcase.v1beta1.DeleteUserRequest;
+import com.google.showcase.v1beta1.EchoSettings;
+import com.google.showcase.v1beta1.IdentityClient;
+import com.google.showcase.v1beta1.IdentitySettings;
+import com.google.showcase.v1beta1.ListUsersRequest;
+import com.google.showcase.v1beta1.ListUsersResponse;
+import com.google.showcase.v1beta1.UpdateUserRequest;
+import com.google.showcase.v1beta1.User;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ITCrud {
+
+  private IdentityClient httpJsonClient;
+
+  @Before
+  public void setup() throws GeneralSecurityException, IOException {
+    // Create HttpJson IdentityClient
+    IdentitySettings httpJsonIdentitySettings =
+        IdentitySettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                EchoSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    httpJsonClient = IdentityClient.create(httpJsonIdentitySettings);
+
+    // Ensure an empty state before each run
+    cleanupData(httpJsonClient);
+  }
+
+  @After
+  public void cleanup() {
+    httpJsonClient.close();
+  }
+
+  private void cleanupData(IdentityClient identityClient) {
+    IdentityClient.ListUsersPagedResponse pagedResponse =
+        identityClient.listUsers(ListUsersRequest.newBuilder().setPageSize(5).build());
+    for (IdentityClient.ListUsersPage listUsersPage : pagedResponse.iteratePages()) {
+      for (User user : listUsersPage.getResponse().getUsersList()) {
+        identityClient.deleteUser(user.getName());
+      }
+    }
+  }
+
+  // This test runs through the four CRUD operations. The operations are
+  // set to build off each other and live inside this one test case.
+  // The tests run in the order of: Create -> List -> Update -> Delete
```

Comment:
Although all living in one test case, splitting into sub-methods seems easily doable with the current structure, and we'd get more information about what stage failed in the stack trace.

Bonus: Assertion would be nice before the CRUD to verify cleanupData was successful.
```java
@Test
public void testHttpJson_CRUD() {
    assert(cleanupData was successful and no users currently exist);

    testHttpJson_Create();
    testHttpJson_Read();
    testHttpJson_Update();
    testHttpJson_Delete();
}

Reply:
Makes sense. I originally had each RPC test individually but I saw that I was making additional RPC calls just to setup each individual test case.
i.e.
- Read, Update, Delete use Create to setup the data 
- Create, Read, Update use Read and Delete to delete the data

> Bonus: Assertion would be nice before the CRUD to verify cleanupData was successful.

I didn't add this in this case since the flow would delete at the end (last RPC in the flow is DELETE) and `cleanUpData()` would basically do nothing (list nothing and delete nothing). The final assertion at the end of the flow was to verify/ assert that the delete worked.

I can change this to test each individual RPC